application/src/test/resources/org/opentripplanner/apis/gtfs/expectations/patterns.json

```diff
@@ -54,6 +54,56 @@
             "occupancy": {
               "occupancyStatus": "FEW_SEATS_AVAILABLE"
             }
+          },
+          {
+            "gtfsId": "F:321Canceled",
+            "stoptimes": [
+              {
+                "stop": {
+                  "gtfsId": "F:Stop_0",
+                  "name": "Stop_0"
+                },
+                "headsign": "Stop headsign at stop 10",
+                "scheduledArrival": 41400,
+                "scheduledDeparture": 41400,
+                "stopPosition": 10,
+                "stopPositionInPattern": 0,
+                "realtimeState": "CANCELED",
```

Comment:
I am now working on making RealTimeTripTimes immutable and I believe this test case is wrong. The `stoptimes` field refers to the scheduled timetable, without real time information.

In the test code, you generated a RealTimeTripTimes object (which was mutable) and put it into the scheduled timetable, then mutated that object by setting it as cancelled, affecting the API output from the scheduled timetable.

Reply:
I have fixed it in my latest PR as it failed the test.